### ISSUE TITLE
Batch inference

### ISSUE THREAD

**Issue by wuyuanyi135** [WXinlong/SOLO#110]
Hello,
Thanks for the great project. I am trying to adapt SOLOv2 into my project and I would like to use batch inference to use my GPU more efficiently. Currently, I noticed these limitations 
In file `mmdet\apis\inference.py`, although the docstring specify that the `img` could be a list of instances, when passing in the list argument, the `data = test_pipeline(data)` will fail because it does not support batched input.

I tried to bypass this limitation by writing a wrapper to map the pipeline to each image and it success. Next, I tried to batch the tensors by doing this dirty hack
`data['img'] = [torch.cat([d_each['img'][0] for d_each in d], dim=0)]`
I was able to get a batched tensor but in `mmdet\models\detectors\base.py`#124 the assertion prevents me to proceed
`assert imgs_per_gpu == 1`

I wonder whether it is possible to perform batched inference with the current codebase to get the inference speed on par to the reported speed?

Regards,
YW

**Comment by WXinlong** [comment]
@wuyuanyi135  It is because that mmdetection didn't support batch inference. But I noticed that they recently support it as in their [changelog](https://github.com/open-mmlab/mmdetection/blob/09fe65ac4be977eefcf7d869cc4dff49db1b701d/docs/changelog.md). You can refer to their updates. I also provided the [SOLO implementation in mmdet v2](https://github.com/WXinlong/mmdetection/tree/SOLO), which may help.

**Comment by wuyuanyi135** [comment]
Thanks for your response 👍
